### DISCUSSION TITLE
How to get the instance of MediaCaptionsRadioGroupElement?

### DISCUSSION THREAD

**Discussion by stsrki** [vidstack/player#1359] [Q&A]
Is there a way to get the instance of MediaCaptionsRadioGroupElement? But without the use of `querySelector`?

This is what I can find in the documentation, but it doesn't work for me:

```js
const instance = document.querySelector("media-captions-radio-group");
```

But I would like to access it directly. If is is possible:

```js
player.something.something.mediaCaptionsElement
```

**Comment by mihar-22** [comment]
It's not accessible atm in the Default Layout, you'll need to build your own layout to get a reference to it.

**Reply by stsrki** [reply]
Is there a plan to add it in the future?

**Reply by mihar-22** [reply]
Not really unless there's a specific reason. Why are you trying to get a reference to it?

**Reply by stsrki** [reply]
I'm converting from Plyr, and in Plyr, we got the `settings` API to control the menu items in the settings menu.

![image](https://github.com/user-attachments/assets/20f47766-d85e-4b65-a003-4dedeb927760)

But in VidStack, this is not available, so my idea is to make it by getting the instances of those items and then show or hide them when needed.

**Reply by mihar-22** [reply]
You can use CSS. Inspect the DOM to find the relevant classes and simply hide what you don't want:

```css
.vds-captions-menu {
  display: none;
}
```

**Reply by stsrki** [reply]
I will try that. Thanks.

Can I ask if you would be able to add a new `settings` API to the PlyrLayout? Similar to what we already have with `controls` API.

![image](https://github.com/user-attachments/assets/36ac90fb-9fa5-426a-911e-08449963e73f)
